### PR TITLE
Maintenance work

### PR DESCRIPTION
**Pull request by greg0ire**
Todo for the maintainers:

- [x] rename `master` to `2.3.x` (if that's indeed what should be done)
- [x] tweak settings to change the required jobs
- [x] merge this
- [x] merge up `2.2.x` into the new `2.3.x`

Also, I see that there is still a `.travis.yml`, I left it but it does not seem to be in use in PRs.

### CODE REVIEW

File: .doctrine-project.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
     "versions": [
         {
             "name": "2.2",
-            "branchName": "master",
+            "branchName": "2.3.x",
```
Comment (SenseException) [inline .doctrine-project.json:10]: Then the name should be `2.3`, right? 2.2 should get its own entry by now.

**Review by SenseException — DISMISSED** [review]

File: .doctrine-project.json (hunk 1 above)
Comment (malarzm) [reply .doctrine-project.json:10]: From what I saw in the past (last release) the changes to `.doctrine-project.json` were applied to `master` after the release. There indeed only branchName would need to change: https://github.com/doctrine/mongodb-odm/blob/4269eea27ffdbdc77c07161d59b8fbc06f1ce39d/.doctrine-project.json#L9-L12

Comment (greg0ire) [reply .doctrine-project.json:10]: Ah right, a different workflow is used on the ODM repos. I'll do the changes here anyway so as to have only one PR and let the changes bubble up with a merge up, WDYT?

Comment (malarzm) [reply .doctrine-project.json:10]: FWIW we can change this, I just had no clue what to do and followed what was done last time :)

**Comment by malarzm** [comment]
@greg0ire could you please elaborate on "tweak settings to change the required jobs" part? Is this about travis that somehow still tries to run?

**Comment by greg0ire** [comment]
See the "Some checks haven't completed yet" at the bottom? They mention 2 jobs that have different names now. That can and should be changed at https://github.com/doctrine/mongodb-odm/settings/branch_protection_rules/16523510

**Comment by greg0ire** [comment]
Alternatively, if the PR is fine with you, approve it and I will take care of the rest.

**Comment by malarzm** [comment]
Ah I see it now, removed them altogether. We're not going to merge with fails anyway 👍

**Review by malarzm — APPROVED** [review]

**Comment by greg0ire** [comment]
Taking your approval as a go ahead, let's do this!

**Comment by greg0ire** [comment]
Ah I see you're doing it already, I'll let you proceed then :)

**Comment by malarzm** [comment]
Ah yeah, I just started clicking since I have some time on my hands :) Prolly we should have coordinated this on Slack :D

**Comment by malarzm** [comment]
Thanks @greg0ire!

**Pull request merged**